src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/SourceTelemetry.cs

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using NuGet.Common;
+using NuGet.Configuration;
+
+namespace NuGet.PackageManagement.Telemetry
+{
+    public static class SourceTelemetry
+    {
+        [Flags]
+        private enum HttpStyle
+        {
+            NotPresent = 0,
+            YesV2 = 1,
+            YesV3 = 2,
+            YesV3AndV2 = YesV3 | YesV2,
+        }
+
+        private static readonly Lazy<string> ExpectedVsOfflinePackagesPath = new Lazy<string>(() =>
+        {
+            try
+            {
+                var programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+                return Path.Combine(programFiles, "Microsoft SDKs", "NuGetPackages");
```

Comment:
how about VS for mac?

Reply:
Good point. I made the code just return `null` in this case.